src/main/java/com/github/steveice10/mc/protocol/packet/ingame/serverbound/inventory/ServerboundSetBeaconPacket.java

```diff
@@ -7,23 +7,34 @@
 import lombok.Data;
 import lombok.With;
 
+import javax.annotation.Nullable;
 import java.io.IOException;
 
 @Data
 @With
 @AllArgsConstructor
 public class ServerboundSetBeaconPacket implements Packet {
-    private final int primaryEffect;
-    private final int secondaryEffect;
+    private final Integer primaryEffect;
```

Comment:
Just make this an OptionalInt - I think your write method here is incorrect anyway.

Reply:
Sure, sorry I didn't know those existed

Reply:
Oh yes, sorry, I was going to do it a different way to begin with and forgot to change the write myself

Reply:
Mark the OptionalInts as Nonnull as well, to comply with the rest of the project.